samcli/local/common/runtime_template.py

```diff
@@ -0,0 +1,98 @@
+"""
+All-in-one metadata about runtimes
+"""
+
+import itertools
+import os
+
+try:
+    import pathlib
+except ImportError:
+    import pathlib2 as pathlib
+
+from samcli.lib.build.workflow_config import PYTHON_PIP_CONFIG, RUBY_BUNDLER_CONFIG,\
+    NODEJS_NPM_CONFIG, JAVA_GRADLE_CONFIG
+
+_init_path = str(pathlib.Path(os.path.dirname(__file__)).parent)
+_templates = os.path.join(_init_path, 'init', 'templates')
+
+# NOTE(TheSriram): Builder configs are only indicative of a builder config that is used for selecting a build workflow.
+# The config mutates based on the workflow selector, so they  are not to be imported and used from the template mapping.
+# They are an indication of metadata associated with a specific runtime.
+
+RUNTIME_DEP_TEMPLATE_MAPPING = {
```

Comment:
This does not make sense. Why would you depend on the build workflow configs within init. They should be treated as isolated modules. Otherwise you will end up with a tangled mess of dependencies one year from now.

Reply:
Will remove that. 👍